packages/polywrap-client-config-builder/polywrap_client_config_builder/client_config_builder.py

```diff
@@ -0,0 +1,256 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from polywrap_core import Uri, IUriResolver, Env, UriWrapper,UriPackage
+from polywrap_uri_resolvers import IUriResolver
+from typing import Dict, Any, List, Optional, Union
+
+@dataclass(slots=True, kw_only=True) 
+class ClientConfig():
+    """
+    This Abstract class is used to configure the polywrap client before it executes a call
+    The ClientConfig class is created and modified with the ClientConfigBuilder module
+    """
+    envs: Dict[Uri, Dict[str, Any]]
+    interfaces: Dict[Uri, List[Uri]]
+    packages: List[UriPackage]
```

Comment:
I believe this can also be optimised to a dict instead of list of object. Can be done in the separate issue though